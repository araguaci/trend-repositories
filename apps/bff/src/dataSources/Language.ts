import { RESTDataSource } from 'apollo-datasource-rest';

export class Language extends RESTDataSource {
  constructor() {
    super();
    this.baseURL =
      'https://raw.githubusercontent.com/github/linguist/master/lib/linguist/';
  }

  async getLanguages() {
    return this.get('languages.yml', null);
  }
}
