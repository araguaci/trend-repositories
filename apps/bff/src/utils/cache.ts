const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
