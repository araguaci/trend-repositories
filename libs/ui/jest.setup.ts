import '@testing-library/jest-dom';
jest.mock('next/dist/client/router', () => require('next-router-mock'));
