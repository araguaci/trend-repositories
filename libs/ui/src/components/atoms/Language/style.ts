import styled from 'styled-components';

export const StyledLanguage = styled.div`
  display: inline-flex;
  align-items: center;
  margin-right: 10px;
`;

export const LanguageColor = styled.div<{ color: string }>`
  position: relative;
  top: 1px;
  display: inline-block;
  width: 12px;
  height: 12px;
  border: 1px solid #f0f6fc33;
  border-radius: 50%;
  margin-right: 5px;
  background-color: ${(props) => props.color};
`;
