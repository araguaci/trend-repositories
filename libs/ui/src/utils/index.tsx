/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
