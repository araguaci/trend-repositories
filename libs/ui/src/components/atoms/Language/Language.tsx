import { Text } from '../../atoms';
import { LanguageColor, StyledLanguage } from './style';
import { FC } from 'react';

interface LanguageProps {
  color?: string;
}

const Language: FC<LanguageProps> = ({ children, color }) => {
  return (
    <StyledLanguage>
      {color && <LanguageColor color={color} />}

      <Text size="small">{children}</Text>
    </StyledLanguage>
  );
};

export default Language;
