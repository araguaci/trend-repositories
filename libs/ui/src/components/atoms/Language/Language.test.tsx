import { renderWithProvider } from '@trend-repositories/ui/utils';
import Language from './Language';

describe('Language Component', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithProvider(
      <Language color="yellow">Javascript</Language>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered without color', () => {
    const { asFragment } = renderWithProvider(<Language>Javascript</Language>);

    expect(asFragment()).toMatchSnapshot();
  });
});
