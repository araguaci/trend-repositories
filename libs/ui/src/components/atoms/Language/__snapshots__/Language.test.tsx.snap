// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Component should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="style__StyledLanguage-x0ke4t-0 fRquPl"
  >
    <div
      class="style__LanguageColor-x0ke4t-1 bXOzcH"
      color="yellow"
    />
    <span
      class="style__StyledText-sc-1aawx67-0 kEyWhB"
    >
      Javascript
    </span>
  </div>
</DocumentFragment>
`;

exports[`Language Component should be rendered without color 1`] = `
<DocumentFragment>
  <div
    class="style__StyledLanguage-x0ke4t-0 fRquPl"
  >
    <span
      class="style__StyledText-sc-1aawx67-0 kEyWhB"
    >
      Javascript
    </span>
  </div>
</DocumentFragment>
`;
